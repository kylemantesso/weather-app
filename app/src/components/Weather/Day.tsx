import React from "react";
import moment from "moment";

import { Box, Text } from "grommet";
import { Icon } from "./Icon";

export const Day: React.FC<{
  icon: string;
  description: string;
  day: string;
  today: boolean;
  max: number;
  min: number;
}> = ({ icon, day, today, max, min, description }) => (
  <Box direction="column" justify="center" align="center" gap="small">
    <Text weight="bold" size="small">
      {today ? "Today" : moment(day).format("dddd")}
    </Text>
    <Icon icon={icon} description={description} />
    <Box direction="row" gap="xsmall">
      <Text weight="bold" size="small">
        {Math.round(max)}°
      </Text>
      <Text size="small">{Math.round(min)}°</Text>
    </Box>
  </Box>
);
