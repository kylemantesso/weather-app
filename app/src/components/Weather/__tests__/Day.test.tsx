import React from "react";
import renderer from "react-test-renderer";
import { Day } from "../Day";

describe("Day", () => {
  it("matches snapshot today", () => {
    const component = renderer.create(
      <Day
        description="Light rain"
        icon="lr"
        max={30}
        min={20}
        day={"2020-02-19"}
        today={true}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("matches snapshot not today", () => {
    const component = renderer.create(
      <Day
        description="Light rain"
        icon="lr"
        max={30}
        min={20}
        day={"2020-02-19"}
        today={true}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
