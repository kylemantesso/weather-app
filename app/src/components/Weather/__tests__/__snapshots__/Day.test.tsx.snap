// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day matches snapshot not today 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 fHyGHh"
>
  <span
    className="StyledText-sc-1sadyjn-0 bDBJwM"
    size="small"
  >
    Today
  </span>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
  />
  <img
    alt="Light rain"
    height="60px"
    src="https://www.metaweather.com/static/img/weather/lr.svg"
  />
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 bIdaiR"
  >
    <span
      className="StyledText-sc-1sadyjn-0 bDBJwM"
      size="small"
    >
      30
      째
    </span>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPZcyD"
    />
    <span
      className="StyledText-sc-1sadyjn-0 hncrgS"
      size="small"
    >
      20
      째
    </span>
  </div>
</div>
`;

exports[`Day matches snapshot today 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 fHyGHh"
>
  <span
    className="StyledText-sc-1sadyjn-0 bDBJwM"
    size="small"
  >
    Today
  </span>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
  />
  <img
    alt="Light rain"
    height="60px"
    src="https://www.metaweather.com/static/img/weather/lr.svg"
  />
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
  />
  <div
    className="StyledBox-sc-13pk1d4-0 bIdaiR"
  >
    <span
      className="StyledText-sc-1sadyjn-0 bDBJwM"
      size="small"
    >
      30
      째
    </span>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 cPZcyD"
    />
    <span
      className="StyledText-sc-1sadyjn-0 hncrgS"
      size="small"
    >
      20
      째
    </span>
  </div>
</div>
`;
